### DISCUSSION TITLE
20221111

### DISCUSSION THREAD

**Discussion by tmiw** [ARSFI/paclink-open-source#34] [General]
This is a pre-production build intended to gauge current status of the project and determine what needs to be fixed. Brief testing indicates that packet (using Kenwood TM-D700A), AGW (using AGWPE and UZ7HO soundmodem), Telnet and Pactor channels work.

Contains (vs. [last pre-release version](https://github.com/ARSFI/paclink-open-source/releases/tag/nightly_20221011)):

* Internal changes in preparation for supporting MAUI-based UI and Linux/Mac port.

Instructions:

1. Unzip paclink-open-source.zip somewhere on your hard drive.
2. Run setup.exe to install on your system. Paclink will be installed in the Start menu.
3. Open Paclink from the Start menu and reconfigure if required. (There is no import functionality from the existing Paclink at this time but if you used a pre-release version, configuration should be saved.)

Please leave comments in the release's discussion thread or email K6AQ@winlink.org with feedback. 👍

<hr /><em>This discussion was created from the release <a href='https://github.com/ARSFI/paclink-open-source/releases/tag/nightly_20221111'>20221111</a>.</em>

**Comment by tmiw** [comment]
@k0qed, we should now be ready to begin creating a MAUI based UI for the project. In theory we just need to create a Paclink.UI.MAUI project (and a .sln file based on the existing one that doesn't include Paclink.UI.Windows) and begin implementing the needed forms. 

Additionally, we should probably implement a minimum set of screens (main form, station setup and Telnet) to help with onboarding additional developers.

**Comment by tmiw** [comment]
@kd4vrf, if you don't mind, can you give this version a shot and make sure there's nothing obvious wrong? Especially on the Pactor side as that's where a lot of the changes I did for this release were in.

**Comment by kd4vrf** [comment]
@tmiw, I got it installed yesterday. All of my setting were gone, I'm not sure if that was from the new software or if I blew them away as a troubleshooting step last time I used it. I use Winlink, it's been awhile since I tried Paclink.

I'll get everything configured again and give it a try and let you know how it goes.

**Reply by tmiw** [reply]
Sounds good, thanks!

**Comment by Spirarementum** [comment]
May I report here the issues I found while testing?

**Reply by tmiw** [reply]
Go for it! What's up?

**Comment by Spirarementum** [comment]
First, thanks for working on this. I am looking forward to this working on Linux. I currently rely heavily upon RMS Express and need to have it running inside a W10 virtual box as my navigation system is exclusively running on Linux. 

Here are a few observations from my test this morning:

1. Telnet connection works fine;
2. Pactor connection does not...see attachments;
3. Also no Ten-Tec TT-588 radio control is available in the drop-down list...so I could not test it.

I was never able to use the alternative option of the radio control via TNC with the PTCusb-II on RMS Express so I suspect the PTCusb-II does not like the TT-588

Let me know if you need me to perform other tests.

[Paclink.log](https://github.com/ARSFI/paclink-open-source/files/11458656/Paclink.log)
![New Bitmap Image](https://github.com/ARSFI/paclink-open-source/assets/73918322/01daf2bd-0a51-4e57-96d9-87ffa2011df4)

Serge
Sailing vessel Spirare

**Reply by tmiw** [reply]
Does "manual" mode work for Pactor? You'd likely have to adjust frequency yourself (not ideal) but the PTC IIusb should at least trigger PTT via the radio cable plugged into it.

**Comment by Spirarementum** [comment]
No it does not.
But the same setting on the same W10 VM are working with RMS Express so it looks like the problem is with Paclink.

`2023-05-12 13:12:43.9005 | DEBUG | Main | *** Initializing the PTC IIusb TNC
2023-05-12 13:12:43.9247 | DEBUG | Main | *** PTC Fault
2023-05-12 13:12:43.9247 | DEBUG | Main | *** Error during PTC IIusb startup... See error log
2023-05-12 13:12:43.9541 | DEBUG | Main | *** TNC PTC IIusb on serial port COM3 initialization failed
2023-05-12 13:12:43.9620 | DEBUG | Main | *** Closing Pactor at 2023/05/12 01:12:43`

**Reply by tmiw** [reply]
Can you look in your C:\Users\[username]\AppData\Roaming\Paclink folder and attach the contents of the Logs and ErrorLogs folders here? You can zip those two up into one ZIP file if that makes it easier.

**Reply by Spirarementum** [reply]
[Paclink.202305121314.log](https://github.com/ARSFI/paclink-open-source/files/11469056/Paclink.202305121314.log)
[Paclink.log](https://github.com/ARSFI/paclink-open-source/files/11469057/Paclink.log)

**Reply by tmiw** [reply]
Thanks. Based on what I'm seeing in the error log, I'm wondering if you need to explicitly specify a TNC configuration file in the "Pactor TNC Channels" dialog (note bottom setting in one of your previous screenshots). If you click Browse, there should be a "TNCConfigurations" folder inside where Paclink got installed; try selecting the "ExamplePTCII_e.aps" or "ExamplePTCII_pro.aps" file and seeing if that behaves any better.

**Reply by Spirarementum** [reply]
Obviously, something is missing:

![1](https://github.com/ARSFI/paclink-open-source/assets/73918322/f84967d8-8770-4896-875f-243e5d81777f)

**Reply by tmiw** [reply]
The TNCConfigurations folder would be in your C:\Program Files\Paclink (or C:\Program Files (x86)\Paclink depending on where it got installed). If you still can't find it, I can attach the files here.

**Reply by Spirarementum** [reply]
Thanks, found the TNC configuration file under \Program Files (x86)\Winlink\Paclink.
Happy to report that I was able to connect via TNC !
Please do let me know if I can help test the TenTec TT-588 remote control once you get to this.

One last comment if I may. This version does not adjust the size of its fixed windows according to the W10 Display Setting / Scale and Layout. With the advent of the availability of small high-definition displays (ex.: 15 or 17 inches 4K),  Scaling above 100% is required otherwise characters are too small to be readable. I am mentioning this because it is an issue with most ham radio apps running either under Windows or a Linux distribution.  The situation gets worst when trying to run a Windows app under Wine hence why I am still stuck with having to run a W10 in a virtual box for RMS Express, Vara, Varim,Fldigi...etc. OpenCPN have recently addressed this under 5.8v and vARIM is still a work in progress.

**Reply by tmiw** [reply]
I made some changes that should hopefully help with the DPI scaling issues. Try https://github.com/ARSFI/paclink-open-source/releases/tag/nightly_20230515 and let me know how it goes?
